src/main/java/org/testng/internal/PoolService.java

```diff
@@ -51,7 +51,7 @@ public Thread newThread(@Nonnull Runnable r) {
     for (Future<FutureType> take : takes) {
       try {
         result.add(take.get());
-      } catch (InterruptedException | ExecutionException e) {
+      } catch (InterruptedException | ExecutionException e) { //NOSONAR
```

Comment:
What was the sonar error?

Reply:
Quoting SonarCube 
`"InterruptedException" should not be ignored.`

> InterruptedExceptions should never be ignored in the code, and simply logging the exception counts in this case as "ignoring". The throwing of the InterruptedException clears the interrupted state of the Thread, so if the exception is not handled properly the fact that the thread was interrupted will be lost. Instead, InterruptedExceptions should either be rethrown - immediately or after cleaning up the method's state - or the thread should be re-interrupted by calling Thread.interrupt() even if this is supposed to be a single-threaded application. Any other course of action risks delaying thread shutdown and loses the information that the thread was interrupted - probably without finishing its task.

Reply:
It looks like a false positive from Sonar as we are rethrowing the exception. You should open an issue on https://community.sonarsource.com/c/bug